crates/turborepo-lib/src/gitignore.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+};
+
+use turbopath::AbsoluteSystemPathBuf;
+
+const TURBO_GITIGNORE_ENTRY: &str = ".turbo";
+const GITIGNORE_FILE: &str = ".gitignore";
+
+pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPathBuf) -> Result<(), io::Error> {
+    let gitignore_path = repo_root.join_component(GITIGNORE_FILE);
+
+    if !gitignore_path.exists() {
+        let mut gitignore = File::create(gitignore_path)?;
+        #[cfg(unix)]
+        gitignore.metadata()?.permissions().set_mode(0o0644);
+        writeln!(gitignore, "{}", TURBO_GITIGNORE_ENTRY)?;
+    } else {
+        let gitignore = File::open(&gitignore_path)?;
+        let mut lines = io::BufReader::new(gitignore).lines();
+        let has_turbo =
+            lines.any(|line| line.map_or(false, |line| line.trim() == TURBO_GITIGNORE_ENTRY));
+        if !has_turbo {
+            let mut gitignore = OpenOptions::new()
+                .read(true)
+                .append(true)
+                .open(&gitignore_path)?;
+
+            // write with a preceding newline just in case the .gitignore file doesn't end
+            // with a newline
+            writeln!(gitignore, "\n{}", TURBO_GITIGNORE_ENTRY)?;
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use anyhow::Result;
+    use tempfile::tempdir;
+
+    use super::*;
+
+    fn has_turbo_gitignore_entry(gitignore_path: &AbsoluteSystemPathBuf) -> bool {
+        let gitignore = File::open(gitignore_path).expect("Failed to open .gitignore file");
+        let mut lines = io::BufReader::new(gitignore).lines();
+        lines.any(|line| line.map_or(false, |line| line.trim() == TURBO_GITIGNORE_ENTRY))
```

Comment:
I see this line duped between this function and the actual implementation. It might be worth pulling out into a helper function to make sure they don't drift.